apps/prairielearn/src/batched-migrations/20250501105324_ai_grading_job_prompt_backfill.sql

```diff
@@ -0,0 +1,28 @@
+-- BLOCK select_bounds
+SELECT
+  min(id),
+  max(id)
+FROM
+  ai_grading_jobs;
+
+-- BLOCK update_ai_grading_job_prompt_type
+UPDATE ai_grading_jobs
+SET
+  prompt = (
+    SELECT
+      jsonb_agg(elem::jsonb)
+    FROM
+      jsonb_array_elements_text(prompt) AS elem
+  )
+WHERE
+  jsonb_typeof(prompt) = 'array'
+  AND NOT EXISTS (
+    SELECT
+      1
+    FROM
+      jsonb_array_elements(prompt) AS elem
+    WHERE
+      jsonb_typeof(elem) != 'string'
+  )
```

Comment:
Nit: in practice this should be equivalent, but it feels more correct to me to process any row where at least one string element exists.

```suggestion
  AND EXISTS (
    SELECT
      1
    FROM
      jsonb_array_elements(prompt) AS elem
    WHERE
      jsonb_typeof(elem) = 'string'
  )
```

Reply:
Resolved in https://github.com/PrairieLearn/PrairieLearn/pull/11889/commits/eec550816bddef71148e9cc9cbda8b1e8e631feb